joram.py

```diff
@@ -44,23 +42,28 @@ def get_pdf_content_lines(pdf_raw_data):
                 yield line
 
 
-def read_pdf_prices(url):
-    """
-    Reads a pdf line by line and verify if contains a gas price.
-    The prices are always in the same order just need to make a match.
-    """
+# Function to extract the gas prices from the PDFs
+def read_pdf_prices(joram_current_year_url):
     discovered_prices = 0
-    response = requests.get(url)
+    response = requests.get(joram_current_year_url)
     for line in get_pdf_content_lines(response.content):
         if discovered_prices == 3:
             break
-        matches = re.search(PDF_GAS_PRICE_REGEX, line)
-        if matches:
+        match = re.search(PDF_GAS_PRICE_REGEX, line)
+        if match:
             discovered_prices += 1
-            yield matches.groups()
+            yield match.groups()
 
 
 if __name__ == "__main__":
-    print(dict(read_pdf_prices(newest_pdf_joram)))
-    # FIXME (DEBUG): Print the hardcoded link
-    print(dict(read_pdf_prices(debug_pdf_joram)))
+    # Loop through the sorted PDF links and find the gas prices
+    while len(sorted_pdf_links) > 0:
+        newest_pdf_link = sorted_pdf_links.pop()
+        newest_pdf_filename = newest_pdf_link["href"].split("/")[-1]
+        newest_pdf_joram = joram_url + current_year + "/" + newest_pdf_filename
```

Comment:
I think that we just need to guarantee that the newest pdf it's from the week we're looking for, to prevent finding and publishing the gas prices from the previous week. Am I thinking right, @HarryVasanth?

Reply:
Yes. For that, we only publish if the price has changed, plus we'd need to extract the dates. 
So if one of you guys will integrate this, then these things needs to be considered. 😅